api/src/permissions/modules/process-ast/lib/inject-cases.ts

```diff
@@ -48,8 +44,12 @@ function processChildren(
 			throw new Error(`Cannot extract access permissions for field "${fieldKey}" in collection "${collection}"`);
 		}
 
-		// Global and field can't both be undefined as per the error check prior
-		child.whenCase = [...(globalWhenCase ?? []), ...(fieldWhenCase ?? [])];
+		// If there's one or more permissions that allow full access to this field, we can safe some
+		// query perf overhead by ignoring the whole case/when system
```

Comment:
I'd probably slightly adjust this comment, for example as follows, as the conditions have now been reversed
```suggestion
		// The case/when system only needs to take place if no full access is given on this field,
		// otherwise we can skip and thus safe some query perf overhead
```